syntax = "proto3";
package cliente_gateway;

message ClienteRequisicao {
  enum TipoRequisicao {
    REQUISICAO_DESCONHECIDA = 0;
    LISTAR_DISPOSITIVOS = 1;
    ENVIAR_COMANDO = 2;
  }

  TipoRequisicao request_type = 1;
  ComandoCliente comando_cliente = 2;  // Renomeado para comando_cliente
}

message ComandoCliente {  // Renomeado para ComandoCliente
  enum TipoComando {
    COMANDO_DESCONHECIDO = 0;
    LIGAR_DESLIGAR = 1;      // Padronizado com gateway-atuadores
    DEFINIR_TEMPERATURA = 2;  // Padronizado com gateway-atuadores
  }

  string id_dispositivo = 1;  // Corrigido o nome do campo
  TipoComando tipo_comando = 2;  // Padronizado o nome
  string valor = 3;  // Padronizado o nome
}

message RespostaGateway {
  enum TipoRequisicao {
    REQUISICAO_DESCONHECIDA = 0;
    LISTAR_DISPOSITIVOS = 1;
    ENVIAR_COMANDO = 2;
    ERRO = 3;
  }

  TipoRequisicao response_type = 1;
  bool sucesso = 2;
  string message = 3;
  repeated DispositivoInfo devices = 4;
}

message DispositivoInfo {
  string id = 1;
  string tipo = 2;
  string ip = 3;
  int32 porta = 4;
  string estado = 5;
}
